src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt

```diff
@@ -11,61 +12,108 @@ import org.slf4j.LoggerFactory
 import java.io.File
 import java.io.InputStream
 import java.nio.charset.Charset
-import java.util.HashMap
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+import java.util.regex.Pattern
+import java.util.stream.Collectors
 import kotlin.concurrent.thread
 
 /**
  * This class takes a path to a docker-compose yaml file and copies it to a temporary location when initialized.
  *
  * After initialization it is possible to start, stop, inspect and kill the containers that are configured in the yaml file.
  */
-class DockerCompose(classPathYmlResource: String,
-                    private val prefix: String,
-                    private val env: HashMap<String, String>) {
+class DockerCompose private constructor(private val prefix: String,
+                                        private val env: Map<String, String>,
+                                        private val dockerComposeFilename: String) {
 
-    val LOGGER: Logger = LoggerFactory.getLogger(DockerCompose::class.java)
-
-    val EXEC_INFO_LOGGER = { line: String ->
-        LOGGER.info(line)
+    init {
+        verifyDockerComposeInstallation()
     }
 
-    val STDOUT_LOG_CONSUMER = { line: String ->
-        LOGGER.debug("OUT: " + line)
-    }
+    constructor(classPathYmlResource: String, applicationPrefix: String, env: Map<String, String>) : this(applicationPrefix, env, initializeNewDockerComposeSession(classPathYmlResource, env))
 
-    val STDERR_LOG_CONSUMER = { line: String ->
-        LOGGER.info("ERR: " + line)
+    constructor(applicationPrefix: String, fileNamePrefix: String) : this(applicationPrefix, loadEnvironmentFile(fileNamePrefix), fileNamePrefix + YAML_FILE_EXTENSION)
+
+    constructor(classPathYmlResource: String, applicationPrefix: String, fileNamePrefix: String, env: Map<String, String>) : this(applicationPrefix, env, fileNamePrefix + YAML_FILE_EXTENSION) {
+        saveResourceToTmpFolder(classPathYmlResource, File(fileNamePrefix + YAML_FILE_EXTENSION))
+        saveEnvironmentFile(env, fileNamePrefix + ENVIRONMENT_FILE_EXTENSION)
     }
 
+    companion object {
```

Comment:
IMO you should put the companion object on top, even before the constructors.

Another question though, is if you need it at all. 
The constants could be placed outside over the class, and the methods could be placed outside, but under the class.

It is possible to have a private fun outside of the class then:

"If you mark a declaration private, it will only be visible inside the file containing the declaration;"

Reply:
I generally don't like polluting the global namespace with top level declarations, and prefer to keep these details scoped to the class. However, as you can guess, I am not an expert in idiomatic Kotlin, so I can change them if you consider it to be a better approach.